### PR TITLE
Fix the issue that rawkv restore's end key is incorrectly inclusive when restoring partial of backed up data

### PR DESCRIPTION
**Pull request by MyonKeminta**
### What problem does this PR solve? 

This PR fixes the issue that rawkv restore's end key is incorrectly inclusive when restoring partial of backed up data. See https://github.com/tikv/tikv/issues/7163

### What is changed and how it works?

This PR adds is_raw_kv field to DownloadRequest and adds end_key_exclusive field to SstMeta. For SST files that contains the restoring end key in its range, it will cut the file's end key to the restoring endKey, and set end_key_exclusive.

- [x] kvproto: https://github.com/pingcap/kvproto/pull/581
- [x] tikv: https://github.com/tikv/tikv/pull/7196

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported variable/fields change
  - Added is_raw_kv to kvproto `DownlaodRequest`
  - Added end_key_exclusive to kvproto `SstMeta`

Side effects

\-

Related changes

 - Need to cherry-pick to the release branch
  - release-3.1 and release-4.0

### Release note

* Fix the issue that in rawkv mode, the end key of restoring range may be included, which is expected to be exclusive.

### CODE REVIEW

**Comment by MyonKeminta** [comment]
@3pointer @kennytm  PTAL thanks!

File: pkg/restore/import.go [hunk 1]
```diff
@@ -397,9 +397,9 @@ func (importer *FileImporter) downloadRawKVSST(
 	if bytes.Compare(importer.rawStartKey, sstMeta.Range.GetStart()) > 0 {
 		sstMeta.Range.Start = importer.rawStartKey
 	}
-	// TODO: importer.RawEndKey is exclusive but sstMeta.Range.End is inclusive. How to exclude importer.RawEndKey?
-	if len(importer.rawEndKey) > 0 && bytes.Compare(importer.rawEndKey, sstMeta.Range.GetEnd()) < 0 {
+	if len(importer.rawEndKey) > 0 && bytes.Compare(importer.rawEndKey, sstMeta.Range.GetEnd()) <= 0 {
```
Comment (3pointer) [inline pkg/restore/import.go:400]: Is it possible `sstMeta.Range.GetEnd()` is empty?

Comment (MyonKeminta) [reply pkg/restore/import.go:400]: It's inclusive so I think will not be empty, but I'm not sure. Maybe I'd better check it here.

**Comment by MyonKeminta** [comment]
@3pointer @kennytm PTAL can we merge it?

**Pull request merged**